File: problems/tier2-intermediate/010-ride-surge-pricing/boilerplate/python/part2/guided.py
# Data class (given).
class PricingContext:
    def __init__(self, baseFare, availableDrivers, activeRideRequests, timeOfDay, weather):
        self.baseFare = baseFare
        self.availableDrivers = availableDrivers
        self.activeRideRequests = activeRideRequests
        self.timeOfDay = timeOfDay
        self.weather = weather

class RideRequest:
    def __init__(self, userId, pickup, dropoff, rideType):
        self.userId = userId
        self.pickup = pickup
        self.dropoff = dropoff
        self.rideType = rideType

# HINT: introduce an abstraction so new variants don't change existing code.

# HINT: start from what this must return, then work backwards to the state it needs.
def calculateSurge(ctx):
    # TODO: write your solution
    return None

# HINT: start from what this must return, then work backwards to the state it needs.
def calculateFare(req, ctx):
    # TODO: write your solution
    return None

File: problems/tier2-intermediate/010-ride-surge-pricing/boilerplate/java/part2/Guided.java
import java.util.*;

// Data class (given).
class PricingContext {
    public double baseFare;
    public int availableDrivers;
    public int activeRideRequests;
    public String timeOfDay;
    public String weather;

    public PricingContext(double baseFare, int availableDrivers, int activeRideRequests, String timeOfDay, String weather) {
        this.baseFare = baseFare;
        this.availableDrivers = availableDrivers;
        this.activeRideRequests = activeRideRequests;
        this.timeOfDay = timeOfDay;
        this.weather = weather;
    }
}

class RideRequest {
    public String userId;
    public String pickup;
    public String dropoff;
    public String rideType;

    public RideRequest(String userId, String pickup, String dropoff, String rideType) {
        this.userId = userId;
        this.pickup = pickup;
        this.dropoff = dropoff;
        this.rideType = rideType;
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static double calculateSurge(PricingContext ctx) {
        // TODO: write your solution
        return 0.0;
    }

    // HINT: start from what this must return, then work backwards to the state it needs.
    public static double calculateFare(RideRequest req, PricingContext ctx) {
        // TODO: write your solution
        return 0.0;
    }

}
